feature/experimental/bgp/otg_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,525 @@
+func configureATE(t *testing.T, ateParams *bgpTestParams, connectionType connType, auth authType) gosnappi.Config {
+	t.Helper()
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := gosnappi.NewConfig()
+
+	topo.Ports().Add().SetName(port1.ID())
+	dev := topo.Devices().Add().SetName(ateAttrs.Name)
+	eth := dev.Ethernets().Add().SetName(ateAttrs.Name + ".Eth")
+	eth.Connection().SetPortName(port1.ID())
+	eth.SetMac(ateAttrs.MAC)
+
+	ip := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(ateAttrs.IPv4).SetGateway(dutAttrs.IPv4).SetPrefix(uint32(ateAttrs.IPv4Len))
+
+	bgp := dev.Bgp().SetRouterId(ateAttrs.IPv4)
+	peerBGP := bgp.Ipv4Interfaces().Add().SetIpv4Name(ip.Name()).Peers().Add()
+	peerBGP.SetName(ateAttrs.Name + ".BGP4.peer")
+	peerBGP.Advanced().SetHoldTimeInterval(ateHoldTime)
+	if auth == md5Auth {
+		peerBGP.Advanced().SetMd5Key(authPassword)
+	}
+	peerBGP.SetPeerAddress(ip.Gateway()).SetAsNumber(uint32(ateParams.localAS))
+	if connectionType == connInternal {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.IBGP)
+	} else {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+	}
+
+	return topo
+
+}
+
+// createCeaseAction creates the BGP cease notification action in gosnappi
+func createCeaseAction(t *testing.T) gosnappi.ControlAction {
+	t.Helper()
+	ceaseAction := gosnappi.NewControlAction().SetChoice(gosnappi.ControlActionChoice.PROTOCOL)
+	ceaseAction.Protocol().SetChoice(gosnappi.ActionProtocolChoice.BGP).Bgp().
+		SetChoice(gosnappi.ActionProtocolBgpChoice.NOTIFICATION).
+		Notification().SetNames([]string{}).
+		SetChoice(gosnappi.ActionProtocolBgpNotificationChoice.CUSTOM).
+		Custom().
+		SetCode(6).
+		SetSubcode(6)
+	return ceaseAction
+}
+
+// TestEstablishAndDisconnect Establishes BGP session between DUT and ATE and Verifies
+// abnormal termination of session using notification message:
+func TestEstablishAndDisconnect(t *testing.T) {
+	// DUT configurations.
+	t.Log("Start DUT config load:")
+	dut := ondatra.DUT(t, "dut")
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure Network instance type on DUT
+	t.Log("Configure Network Instance")
+	dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+
+	t.Log("Configure BGP")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+
+	bgpClearConfig(t, dut)
+	dutConf := bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS}, dut)
+	gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
```

Comment:
Should also configure password auth here for both the DUT/ATE to match the ATE TestEstablishAndDisconnect case (which reflects the updated README)

Reply:
@octpetre : This change is still required.

Reply:
Done